### PR TITLE
WWST-4476 - DTH upgrade - BOSH Radion ZigBee Multisensor RFMS-ZBMS-3, DTH refactoring

### PR DESCRIPTION
**Pull request by KKlimczukS**
This PR contains crucial changes that help integrate BOSH Radion ZigBee Multisensor RFMS-ZBMS-3 that have been discussed some time ago here -> 
https://github.com/SmartThingsCommunity/SmartThingsPublic/pull/2978

Device changes its type (dth) depending on pairing method (as a open-close sensor, tilt or water leak sensor). Looks and works quite good in ST Classic and OneApp.
Please take a look at this DTH.
@greens @tpmanley @MMateusiakS @PKacprowiczS @MGoralczykS @ZWozniakS

### CODE REVIEW

File: devicetypes/smartthings/smartsense-open-closed-sensor.src/smartsense-open-closed-sensor.groovy [hunk 1]
```diff
@@ -102,13 +103,21 @@ def parse(String description) {
 				} else {
 					log.warn "IAS ZONE REPORTING CONFIG FAILED- error code: ${descMap.data[0]}"
 				}
+			} else if (descMap?.clusterInt == zigbee.IAS_ZONE_CLUSTER && descMap.attrInt == 0x0001 && isBoschRadionMultiSensor()) {
```
Comment (tpmanley) [inline devicetypes/smartthings/smartsense-open-closed-sensor.src/smartsense-open-closed-sensor.groovy]: Does this work when the device is running locally?

**Review by greens — COMMENTED** [review]
I think the whitespace can be left how it was

File: devicetypes/smartthings/smartsense-open-closed-sensor.src/smartsense-open-closed-sensor.groovy (hunk 1 above)
Comment (KKlimczukS) [reply devicetypes/smartthings/smartsense-open-closed-sensor.src/smartsense-open-closed-sensor.groovy]: Double-checked, works for both configurations -> 
runLocally set to TRUE,  executeCommandsLocally set to FALSE
runLocally set to FALSE, executeCommandsLocally set to FALSE

**Review by greens — DISMISSED** [review]

**Comment by greens** [comment]
Looks good to me, but I'll let @tpmanley take one more look.

File: devicetypes/smartthings/smartsense-open-closed-sensor.src/smartsense-open-closed-sensor.groovy (hunk 1 above)
Comment (tpmanley) [inline devicetypes/smartthings/smartsense-open-closed-sensor.src/smartsense-open-closed-sensor.groovy]: Can you add "constants" for the attribute IDs and the zone types?

File: devicetypes/smartthings/smartsense-open-closed-sensor.src/smartsense-open-closed-sensor.groovy [hunk 2]
```diff
@@ -102,13 +103,21 @@ def parse(String description) {
...
+				if (Integer.parseInt(descMap.value, 16) == 0x0015) {
+					//multi-sensor is in contact or tilt detector mode - both act as contact sensor so no action is necessary
+				} else if (Integer.parseInt(descMap.value, 16) == 0x002A) {
+					//multi-sensor is in water detector mode, DTH should be changed to water sensor DTH
+					log.debug "SmartSense Moisture Sensor"
```
Comment (tpmanley) [inline devicetypes/smartthings/smartsense-open-closed-sensor.src/smartsense-open-closed-sensor.groovy]: Can you clarify that the dth is being changed in this log message?

File: devicetypes/smartthings/smartsense-open-closed-sensor.src/smartsense-open-closed-sensor.groovy [hunk 3]
```diff
@@ -102,13 +103,21 @@ def parse(String description) {
...
+					setDeviceType("SmartSense Moisture Sensor")
+				}
 			}
 		}
 	} else if (map.name == "temperature") {
 		if (tempOffset) {
 			map.value = (int) map.value + (int) tempOffset
 		}
-		map.descriptionText = temperatureScale == 'C' ? '{{ device.displayName }} was {{ value }}°C' : '{{ device.displayName }} was {{ value }}°F'
+		map.descriptionText = temperatureScale == 'C' ? '{{ device.displayName }} was {{ value }}�C' : '{{ device.displayName }} was {{ value }}�F'
```
Comment (tpmanley) [inline devicetypes/smartthings/smartsense-open-closed-sensor.src/smartsense-open-closed-sensor.groovy]: This doesn't look like an intentional change

Comment (KKlimczukS) [reply devicetypes/smartthings/smartsense-open-closed-sensor.src/smartsense-open-closed-sensor.groovy]: Thanks for pointing that out, it was "copy-paste & character encoding" issue.

File: devicetypes/smartthings/smartsense-open-closed-sensor.src/smartsense-open-closed-sensor.groovy (hunk 1 above)
Comment (KKlimczukS) [reply devicetypes/smartthings/smartsense-open-closed-sensor.src/smartsense-open-closed-sensor.groovy]: Done. Some more constants were added in other places too.

**Review by tpmanley — APPROVED** [review]

**Pull request merged**